### PR TITLE
Swift call site breakpoints

### PR DESCRIPTION
**Pull request by jimingham**
This is a cherry-pick of the main patch to add the ability to set breakpoints on call-site file & line numbers, and to handle hitting breakpoints in a virtual inlined call stack correctly.

### CODE REVIEW

**Comment by jimingham** [comment]
@swift-ci test

**Comment by jimingham** [comment]
@swift-ci please test

**Comment by jimingham** [comment]
@swift-ci please test

File: lldb/test/API/lang/swift/macro/TestSwiftMacro.py [hunk 1]
```diff
@@ -34,19 +34,58 @@ def testDebugging(self):
...
+        )
 
         self.expect('expression -- #stringify(1)', substrs=['0 = 1', '1 = "1"'])
 
+        # Step out should get us out of stringify, then in to the next macro:
+        thread.StepOut()
+        self.assertIn("a.testStringify", thread.frames[0].name, "Step out back to origin")
+        thread.StepInto()
+        self.assertIn(
+            "freestanding macro expansion #1 of no_return in module a file main.swift line 6 column 3",
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/macro/TestSwiftMacro.py:64]: These demanglings may not be stable — but whoever changes them in the future can also update this test.

**Review by adrian-prantl — APPROVED** [review]

File: lldb/test/API/lang/swift/macro/TestSwiftMacro.py (hunk 1 above)
Comment (jimingham) [reply lldb/test/API/lang/swift/macro/TestSwiftMacro.py:64]: Yeah, it was so handy that these names had the compiler's view of both the call site and the expansion order to confirm we hit the right one, and it serves as a check against our reading of the line table.  So I do think it's worth tracking changes in the demangling.

**Comment by jimingham** [comment]
The Linux bot failed because the "set a breakpoint on a REPL defined function then call it" timed out without so far as I can tell producing anything to match against.  Sounds like the REPL just never came up?  I'll try again.

**Comment by jimingham** [comment]
@swift-ci please test Linux

**Pull request merged**